### PR TITLE
tools: don't parse --force as valid SHA

### LINKED ISSUE #18929: [tools/node-modules] checkout --force doesn't work
### Explain what happens

1. call `./tools/node_modules checkout --force`
2. Output:
  ```
    FETCH    node_modules  [ref: --force]
  fatal: Failed to resolve 'HEAD' as a valid ref.
  ```

### Where is the problem in Cockpit?

[tools/node-modules](https://github.com/cockpit-project/cockpit/blob/main/tools/node-modules)

### PR DESCRIPTION
**Pull request by jelly**
Closes #18929

### CODE REVIEW

File: tools/node-modules [hunk 1]
```diff
@@ -25,7 +25,12 @@ cmd_remove() {
 
 cmd_checkout() {
     # we default to check out the node_modules corresponding to the gitlink in the index
-    local sha="${1-$(get_index_gitlink node_modules)}"
+    local sha
+    if [ "${1-}" != "--force" ]; then
```
Comment (allisonkarlitskaya) [inline tools/node-modules:29]: This feels like the wrong fix...

What if we want `--force` and the SHA?

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
I feel like `shift` somewhere might help us.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Unit tests fail with

    fatal: Failed to resolve '' as a valid ref.

Thanks!

File: tools/node-modules [hunk 2]
```diff
@@ -25,14 +25,20 @@ cmd_remove() {
...
 cmd_checkout() {
     # we default to check out the node_modules corresponding to the gitlink in the index
-    local sha="${1-$(get_index_gitlink node_modules)}"
+    local force=false
+    if [ "${1-}" = "--force" ]; then
+	    force=true
+	    shift
+    fi
+
+    sha="${*-$(get_index_gitlink node_modules)}"
```
[the remark this answers is not in the document]
Comment (jelly) [reply tools/node-modules:34]: No, $1 is unset when you call `shift`. Honestly I find no joy anymore in POSIX shell

[the remark this answers is not in the document]
Comment (jelly) [reply tools/node-modules:34]: Hmm seems I am wrong https://unix.stackexchange.com/questions/174566/what-is-the-purpose-of-using-shift-in-shell-scripts

Either way, POSIX shell is so limiting it :)

[the remark this answers is not in the document]
Comment (martinpitt) [reply tools/node-modules:34]: No, that's precisely the point of `shift` -- `$2` becomes `$1` and so on.

**Review by martinpitt — DISMISSED** [review]
LGTM, thanks!

**Comment by jelly** [comment]
Hmpfft I have seem to have regressed something:

```[jelle@t14s][~/projects/cockpit/tools-node-modules-force]%./tools/node-modules checkout 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1
  REMOVE   node_modules
  CLONE    node_modules  [ref: 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1]
[jelle@t14s][~/projects/cockpit/tools-node-modules-force]%
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[jelle@t14s][~]%cd projects/cockpit/main
[jelle@t14s][~/projects/cockpit/main]%./tools/node-modules checkout 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1
package.json - differ: byte 255, line 7

*** node_modules 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1 doesn't match our package.json
*** refusing to automatically check out node_modules

Options:

    - tools/node-modules checkout --force     # disable this check

    - tools/node-modules install              # npm install with our package.json

./tools/node-modules: *** aborting

```

**Comment by jelly** [comment]
POSIX sh is hard....
```
[jelle@t14s][~/projects/cockpit/tools-node-modules-force]%./tools/node-modules checkout 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1
package.json - differ: byte 255, line 7

*** node_modules 7aec4437ed49da19b6a5d4952d9bce60dadbdfb1 doesn't match our package.json
*** refusing to automatically check out node_modules

Options:

    - tools/node-modules checkout --force     # disable this check

    - tools/node-modules install              # npm install with our package.json

./tools/node-modules: *** aborting

[jelle@t14s][~/projects/cockpit/tools-node-modules-force]%./tools/node-modules checkout
  REMOVE   node_modules
  CLONE    node_modules  [ref: afba1086947543932bf771197ee6eac6462b82ce]
```

**Comment by martinpitt** [comment]
Yeah, using `false` and `true` in shell scripts is *highly* unusual. It's more idiomatic to use an empty string and "1" or so, and check with `[ -n "$flag" ]`

**Comment by jelly** [comment]
> 



> Yeah, using `false` and `true` in shell scripts is _highly_ unusual. It's more idiomatic to use an empty string and "1" or so, and check with `[ -n "$flag" ]`

Want me to do that before reviewing?

**Comment by martinpitt** [comment]
I'd prefer that, yes. booleans in shell are *weird* (not even mentioning the fact that 0 is true and 1 is false..)

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**